Library/Homebrew/utils/pypi.rb

```diff
@@ -358,7 +358,7 @@ def self.update_python_resources!(formula, version: nil, package_name: nil, extr
 
     ohai "Updating resource blocks" unless silent
     Utils::Inreplace.inreplace formula.path do |s|
-      if s.inreplace_string.scan(inreplace_regex).length > 1
+      if T.must(s.inreplace_string.split(/^  test do\b/, 2).first).scan(inreplace_regex).length > 1
```

Comment:
Is it genuinely impossible that the `T.must` won't be truthy here?

Reply:
The only way I can think of is if the Formula file is entirely blank. In this case, `s.inreplace_string == ""` and that will lead to nil:
```console
❯ brew ruby -e 'p "".split(/^  test do\b/, 2).first'
nil
```

If the file doesn't exist, `brew` fails earlier due to `File.binread`. If the file has at least a single character then it will return a String:
```console
❯ brew ruby -e 'p " ".split(/^  test do\b/, 2).first'
" "
```

I didn't think that would happen, but if there is an odd situation which allows for it then I can change check to something safer, e.g.
```suggestion
      formula_contents_without_test = s.inreplace_string.split(/^  test do\b/, 2).first
      if formula_contents_without_test.nil? || formula_contents_without_test.scan(inreplace_regex).length > 1
```
```suggestion
      s.inreplace_string.split(/^  test do\b/, 2).first&.scan(inreplace_regex)&.length.to_i > 1
```